src/org/nuxeo/openshift/library/ConnectClient.groovy

```diff
@@ -0,0 +1,87 @@
+package org.nuxeo.openshift.library
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import java.util.ArrayList
+import groovy.lang.Closure
+import org.apache.http.auth.UsernamePasswordCredentials
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.entity.StringEntity
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.impl.client.CloseableHttpClient
+import org.apache.http.impl.client.HttpClientBuilder
+
+import static org.apache.http.auth.AuthScope.ANY
+
+def deployPackageToConnect(connectUsername, connectPassword, studioProject, packageFilePath, connectUrl='https://connect.nuxeo.com') {
+  def clientId = getClientIdByStudioProject(connectUsername, connectPassword, studioProject)
+  if (clientId == null) {
+    error("No clientId found with user '" + connectUsername + "' for Studio project '" + studioProject + "'.").
+    return
+  }
+
+  def uploadUrl = connectUrl + "/nuxeo/site/marketplace/upload?batch=true\\&project=${studioProject}\\&owner=${clientId}\\&client=${clientId}"
+  def format = "CONNECT_HTTP_STATUS: %{http_code}"
+  def packageFileArg = "package=@${packageFilePath}"
+  def connectResponseFile = "connect_response.txt"
+
+  sh "curl -w '${format}' -u ${connectUsername}:${connectPassword} -i -n -F ${packageFileArg} ${uploadUrl} > ${connectResponseFile}"
+  handleResponse(connectResponseFile)
+}
+
+protected getClientIdByStudioProject(connectUsername, connectPassword, studioProject) {
+  def applicableClients = queryConnectApplicableClients(connectUsername, connectPassword)
+  def clientId = findClientIdByStudioProject(applicableClients, studioProject)
+
+  return clientId
+}
+
+protected handleResponse(connectResponseFile) {
+  def httpStatusFile = "http_status.txt"
```

Comment:
I'm pretty sure we can find something more appropriate to read the content of a file

Reply:
@dmetzler 
Improved even more in last commit here:
https://github.com/nuxeo/nuxeo-openshift/pull/10/commits/1124fb9300ecc5d6afa97843c7efbd992364e189